dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs

```diff
@@ -710,34 +741,27 @@ public static class OpenAIServiceCollectionExtensions
     /// Adds the Azure OpenAI chat completion service to the list.
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection"/> instance to augment.</param>
-    /// <param name="deploymentName">Azure OpenAI deployment name, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
-    /// <param name="modelId">OpenAI model name, see https://platform.openai.com/docs/models</param>
-    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="serviceId">A local identifier for the given AI service</param>
+    /// <param name="serviceConfig">Service configuration <see cref="OpenAIServiceConfig"/></param>    /// <param name="serviceId">A local identifier for the given AI service</param>
     /// <returns>The same instance as <paramref name="services"/>.</returns>
     public static IServiceCollection AddAzureOpenAIChatCompletion(
         this IServiceCollection services,
-        string deploymentName,
-        string modelId,
-        string endpoint,
-        string apiKey,
+        OpenAIServiceConfig serviceConfig,
         string? serviceId = null)
     {
         Verify.NotNull(services);
-        Verify.NotNullOrWhiteSpace(deploymentName);
-        Verify.NotNullOrWhiteSpace(endpoint);
-        Verify.NotNullOrWhiteSpace(apiKey);
+        Verify.NotNullOrWhiteSpace(serviceConfig.DeploymentName);
```

Comment:
In similar method above we also have `NotNull` check for `serviceConfig`:
```suggestion
        Verify.NotNull(serviceConfig);
        Verify.NotNullOrWhiteSpace(serviceConfig.DeploymentName);
```